app-modules/knowledge-base/src/Filament/Actions/DraftWithAiAction.php

```diff
@@ -0,0 +1,146 @@
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\KnowledgeBase\Filament\Actions;
+
+use Closure;
+use Filament\Forms\Get;
+use Filament\Forms\Set;
+use Laravel\Pennant\Feature;
+use App\Settings\LicenseSettings;
+use Filament\Resources\Pages\Page;
+use Filament\Support\Enums\MaxWidth;
+use Illuminate\Support\Facades\Vite;
+use AdvisingApp\Ai\Models\AiAssistant;
+use AdvisingApp\Ai\Settings\AiSettings;
+use Filament\Forms\Components\Textarea;
+use Filament\Notifications\Notification;
+use Filament\Forms\Components\Actions\Action;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use AdvisingApp\Ai\Exceptions\MessageResponseException;
+use AdvisingApp\Ai\Settings\AiIntegratedAssistantSettings;
+
+class DraftWithAiAction extends Action
+{
+    protected array | Closure $mergeTags = [];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this
+            ->label('Draft with AI Assistant')
+            ->link()
+            ->icon('heroicon-m-pencil')
+            ->modalContent(fn (Page $livewire) => view('knowledge-base::filament.actions.draft-with-ai-modal-content', [
+                'recordTitle' => $livewire?->data['title'],
+                'avatarUrl' => AiAssistant::query()->where('is_default', true)->first()
+                    ?->getFirstTemporaryUrl(now()->addHour(), 'avatar', 'avatar-height-250px') ?: Vite::asset('resources/images/canyon-ai-headshot.jpg'),
+            ]))
+            ->modalWidth(MaxWidth::ExtraLarge)
+            ->modalSubmitActionLabel('Draft')
+            ->form([
+                Textarea::make('instructions')
+                    ->hiddenLabel()
+                    ->rows(4)
+                    ->placeholder('What do you want to write about?')
+                    ->required(),
+            ])
+            ->action(function (array $data, Get $get, Set $set, Page $livewire) {
+                $service = Feature::active('ai-integrated-assistant-settings')
+                    ? app(AiIntegratedAssistantSettings::class)->default_model->getService()
+                    : app(AiSettings::class)->default_model->getService();
+
+                $userName = auth()->user()->name;
+                $userJobTitle = auth()->user()->job_title ?? 'staff member';
+                $clientName = app(LicenseSettings::class)->data->subscription->clientName;
+                $articleTitle = $livewire?->data['title'];
+
+                try {
+                    $content = $service->complete(
+                        <<<EOL
+                                    My name is {$userName}, and I am a {$userJobTitle} at {$clientName}. I am currently editing an article in the knowledge base of the college. The current title is "{$articleTitle}". 
+
+                                    Please provide the title as a plain string, without any prefixes, labels like "Title:", or formatting such as bold, italics, or quotes. The title should appear as a simple string.
```

Comment:
You may want to be explicit that the title and content should have a newline separation between them like we do for the Engagement prompt